src/index.d.ts

```diff
@@ -76,13 +85,12 @@ export interface Replica<D extends Record<string, unknown> = {}, T extends Recor
 	 * Performs `table.remove(t, index)` where `t` is a numeric sequential array `table` located in `path`.
 	 */
 	ArrayRemove<P extends Path<D>>(path: P, index: number): Array<PathValue<D, P>>[number];
-	//TODO writelib types?
 	/**
 	 * Calls a function within a [WriteLib](https://madstudioroblox.github.io/ReplicaService/api/#writelib)
 	 * that has been assigned to this `Replica`
 	 * for both the server and all clients that have this `Replica` replicated to them.
 	 */
-	Write(functionName: string, ...params: unknown[]): unknown;
+	Write<P extends Path<W>>(functionName: P, ...params: Parameters<OmitFirstArg<PathValue<W, P>>>): ReturnType<PathValue<W, P>>;
```

Comment:
This line seems to be erroring. I'll have to investigate PathValue being excessively deep but the types here are not valid.